### PR TITLE
die when Ansible run doesn't complete before instance shuts down

### LINKED ISSUE #50: add AWS completion semaphore to ec2-create
AWS instances may terminate and return `Connection to ec2-34-237-51-137.compute-1.amazonaws.com closed by remote host.` to ec2-create without throwing an error. Add an AWS completion semaphore to catch this condition.

### PR DESCRIPTION
**Pull request by donsizemore**
closes #50

### CODE REVIEW

File: ec2/ec2-create-instance.sh [hunk 1]
```diff
@@ -221,8 +221,12 @@ sudo yum -q -y install ansible git nano
 git clone -b $DA_BRANCH https://github.com/GlobalDataverseCommunityConsortium/dataverse-ansible.git dataverse
 export ANSIBLE_ROLES_PATH=.
 ansible-playbook $VERBOSE_ARG -i dataverse/inventory dataverse/dataverse.pb --connection=local $GVARG
+touch /tmp/ansible_complete
 EOF
 
+# did AWS go AWOL?
+rsync -av -e "ssh -i $PEM_FILE" centos@$PUBLIC_DNS:/tmp/dataverse/ansible_complete /tmp/ && /bin/rm /tmp/ansible_complete
```
Comment (pdurbin) [inline ec2/ec2-create-instance.sh:228]: Is the idea that if "ansible_complete" is missing that this rsync command will fail, which will result in the build being marked as failed?

File: ec2/ec2-create-instance.sh [hunk 2]
```diff
@@ -221,8 +221,12 @@ sudo yum -q -y install ansible git nano
 git clone -b $DA_BRANCH https://github.com/GlobalDataverseCommunityConsortium/dataverse-ansible.git dataverse
 export ANSIBLE_ROLES_PATH=.
 ansible-playbook $VERBOSE_ARG -i dataverse/inventory dataverse/dataverse.pb --connection=local $GVARG
+touch /tmp/ansible_complete
```
Comment (pdurbin) [inline ec2/ec2-create-instance.sh:224]: I like this idea of the "ansible_complete" file but some of the steps above could fail, right? Should we separate all these commands with `&&` to give us confidence that they all succeeded?

**Review by pdurbin — COMMENTED** [review]
I left a couple comments.

File: ec2/ec2-create-instance.sh (hunk 1 above)
Comment (donsizemore) [reply ec2/ec2-create-instance.sh:228]: yes.

File: ec2/ec2-create-instance.sh (hunk 2 above)
Comment (donsizemore) [reply ec2/ec2-create-instance.sh:224]: we set `-e` in the sh-bang, so in theory these failures should also cause the script to exit. it was the unexpected SSH termination we're working around.

**Review by pdurbin — APPROVED** [review]
Seems like a nice improvement.

**Pull request merged**